kernel/src/process.rs

```diff
@@ -243,6 +243,52 @@ pub trait ProcessType {
     fn debug_syscall_called(&self);
 }
 
+/// Generic trait for implementing process restart policies.
+///
+/// This policy allows a board to specify how the kernel should decide whether
+/// to restart an app after it crashes.
+pub trait ProcessRestartPolicy {
+    /// Decide whether to restart the `process` or not.
+    ///
+    /// Returns `true` if the process should be restarted, `false` otherwise.
+    fn evaluate(&self, process: &dyn ProcessType) -> bool;
```

Comment:
This type signature/name is kind of clunky. If it should return a `bool` (indicating whether the process should be restarted, maybe it the method should be called `should_restart` or something.

Alternatively, maybe call it `fault_behavior` have it return a specialized enum:

```rust
enum FaultBehavior {
  Restart,
  Terminate
}
```